### PR TITLE
Resolve idependent spanning types when comparing

### PR DESCRIPTION
**Pull request by nsmanville**
Issue: #3066

Any feedback is appreciated.

### CODE REVIEW

File: Cython/Compiler/PyrexTypes.py [hunk 1]
```diff
@@ -4529,7 +4529,7 @@ def independent_spanning_type(type1, type2):
             type1 = type1.ref_base_type
         else:
             type2 = type2.ref_base_type
-    if type1 == type2:
+    if type1.resolve() == type2.resolve():
```
Comment (scoder) [inline Cython/Compiler/PyrexTypes.py]: I think this should also apply to the other comparisons below, not just here.

File: tests/compile/ctypedef_bint.pyx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# distutils: extra_compile_args=-Werror
+# mode: compile
```
Comment (scoder) [inline tests/compile/ctypedef_bint.pyx]: Could you please move this file into `tests/run/` and turn it into a runnable test, using a function with (Python) doctest? It's nice that it compiles, but I would also like to be sure that it produces the correct result. :)

Please also add some additional tests that mix `bint` with a regular C `int`, return the result from a function (which should convert it to a Python bool), etc.

**Comment by 0dminnimda** [comment]
@scoder this also should be close now when #4660 is merged

**Comment by scoder** [comment]
Superseded by #4660.

**Pull request closed without merging**